### DISCUSSION TITLE
edge finding

### DISCUSSION THREAD

**Discussion by tcurdt** [grblHAL/core#525] [Q&A]
I was searching the wiki for some information on edge finding.

Manually setting a new origin is one thing but what about rotations?

And is there a way to measure a grid for surface compensation?

**Comment by terjeio** [comment]
Edge finding is by probing commands, [G38.n](https://www.linuxcnc.org/lucid/docs/html/gcode/g-code.html#gcode:g38).

> Manually setting a new origin is one thing but what about rotations?

[G10L2](https://www.linuxcnc.org/lucid/docs/html/gcode/g-code.html#gcode:g10-l2) rotation is not supported but may be added later. For now you have to rely on sender functionality.

> And is there a way to measure a grid for surface compensation?

A grid can be measured by G38.n commands, however collecting the data points and transforming the gcode is something that has to be done by the sender.

**Reply by tcurdt** [reply]
Ah OK, given that Marlin and friends supports bed levelling I assumed there was support - or at least planned.

Applying those transformations on the sender side is probably a bit more work than just passing all movements through a matrix?
This is implemented in iosender, right?

**Reply by terjeio** [reply]
Marlin and friends are for 3D printer controllers, Grbl/grblHAL is for "normal" CNC machines. The grblHAL core supports most of the NIST RS274NGC spec + LinuxCNC extensions (and a few Mach3 gcodes) and I want to keep it that way. Plugins may add support for gcodes (M-codes) that that originate from the 3D printer world though. So if someone want to write a bed levelling plugin I can look into how the core can be expanded to allow transformations.

> Applying those transformations on the sender side is probably a bit more work than just passing all movements through a matrix?

Well, movements has to be identified in the gcode first so kind of yes to that I guess.

> This is implemented in iosender, right?

Yes, and probing and rotation functionality is also available.
